### DISCUSSION TITLE
How to return previous and next record column from where id with one prisma query?

### DISCUSSION THREAD

**Discussion by skorotkiewicz** [prisma/prisma#12173] [Q&A] [answered]
It should return like this: 
```javascript
{
id: "searched id"
prev: "prev record id",
next: "next record id"
}
```

**Comment by nurul3101** [comment, marked as answer]
Hey @skorotkiewicz 👋 

Your use case seems to be similar to pagination.
Are you trying to achieve cursor-based pagination?

We have an example [here](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#example-filtering-and-cursor-based-pagination) where it's demonstrated how to get a previous record id by passing a negative number ( like -1 ) to get the previous record it and a positive number ( like +1 ) to get the next record in the `take` parameter.

**Reply by skorotkiewicz** [reply]
Thank you for your reply. I solved it by using row query.
```javascript
  const next =
    await prisma.$queryRaw`select slug,question from "Question" WHERE id > ${data.id} ORDER BY id ASC LIMIT 1;
  `;
  const prev =
    await prisma.$queryRaw`select slug,question from "Question" WHERE id < ${data.id} ORDER BY id DESC LIMIT 1;
  `;
```

It was important to me that it should display the next or previous id, even if several records are missing from the database.
`1, 2, 4, 5`
When you ask from 2, the next ID is 4.

In Prisma:
```javascript
  const next = await prisma.question.findMany({
    take: 1,
    where: {
      id: {
        gt: data.id,
      },
    },
    orderBy: {
      id: "asc",
    },
  });
  const prev = await prisma.question.findMany({
    take: 1,
    where: {
      id: {
        lt: data.id,
      },
    },
    orderBy: {
      id: "desc",
    },
  });
```

**Reply by ace4port** [reply]
While this solution works well, I wanted to know if it could be done in since query?
Ideally - current record select all details, while for the next and previous one's only some say slug was returned?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
